spyder_notebook/notebookplugin.py

```diff
@@ -375,6 +387,16 @@ def open_notebook(self, filenames=None):
             for filename in filenames:
                 self.create_new_client(filename=filename)
 
+    def open_console(self, client=None):
+        """Open an IPython console for the given client or the current one."""
+        if not client:
+            client = self.get_current_client()
+        if self.ipyconsole is not None:
+            kernel_name = 'kernel-' + client.get_kernel_id()[0] + '.json'
+            kernel_path = osp.join(jupyter_runtime_dir(), kernel_name)
+            self.ipyconsole._create_client_for_kernel(kernel_path, None, None,

```

Comment:
Not working, I think it is because `osp.dirname` is returning an empty string instead of `None`.